src/intersection.ts

```diff
@@ -14,7 +14,7 @@ type Instance = {
 
 const INSTANCE_MAP: Map<Element, Instance> = new Map()
 const OBSERVER_MAP: Map<string, IntersectionObserver> = new Map()
-const ROOT_IDS: WeakMap<Element, string> = new WeakMap()
+const ROOT_IDS: Map<Element, string> = new Map()
```

Comment:
presumably you'd want `(WeakMap | Map)<Element, string> = new (WeakMap || Map)()` here, to get the memory advantages when possible?